frontend/src/metabase/sharing/components/SharingSidebar.jsx

```diff
@@ -285,23 +258,27 @@ class SharingSidebar extends React.Component {
 
     await this.props.saveEditingPulse();
 
-    await this.props.fetchPulsesByDashboardId(dashboard.id);
     this.setState({ editingMode: "list-pulses" });
   };
 
   createSubscription = () => {
-    this.setState({
-      editingMode: "new-pulse",
-      returnMode: this.state.editingMode,
+    this.setState(({ editingMode, returnMode }) => {
+      return {
+        editingMode: "new-pulse",
+        returnMode: returnMode.concat([editingMode]),
+      };
     });
+
     this.props.setEditingPulse(null, null);
   };
 
   editPulse = (pulse, channelType) => {
     this.setPulse(pulse);
-    this.setState({
-      editingMode: "add-edit-" + channelType,
-      returnMode: this.state.editingMode,
+    this.setState(({ editingMode, returnMode }) => {
+      return {
+        editingMode: "add-edit-" + channelType,
+        returnMode: returnMode.concat([editingMode || "list-pulses"]),
```

Comment:
I think `editingMode` might be guaranteed to be present now (when this code was written we explicitly set it to `undefined` at times).